src/penguin.c

```diff
@@ -266,12 +283,16 @@ float *pen_emperor_penguin(obj_func_t obj_func,
           memcpy(&updated_positions[penguin_j * dim * colony_size + n_updates_per_pengu[penguin_j] * dim],
                   spiral, dim * sizeof(float));
           n_updates_per_pengu[penguin_j] += 1;
+
+          // free(spiral);
         }
       }
     }
 
     // accumulate changes for every pengu during this iteration
     for (size_t pengu_idx = 0; pengu_idx < colony_size; pengu_idx++) {
+      // float* mean_pos = (float*)malloc(dim*sizeof(float));
+      float mean_pos[dim];
```

Comment:
@jakobbeckmann 
strange behavior here
this line right here causes the integration test to fail for rastigrin for dim = 4. 
keep in mind the algo behaves normally. just doesnt converge at the given parameters.

however if I use malloc to declare this array, the test magically converges for dim = 4. 
can you reproduce? i.e run integration test with dim=4 for for rastrigin(it should fail). then comment this out and comment in the malloc decl.  and run test again(it should pass)

if you run it in debug mode, you'll see that it makes zero difference. the average and best function value are the same in both cases (as they should be), but the test fails in one case and doesn't in another.

Reply:
Yeah the whole C thing is freaking weird fml. I also have the `pengu_integration::rosenbrock` integration test that consistently fails on my mac, but on my linux machine is consistently passes and same on travis ...

Maybe we should see with @matthaeusheer , since we are optimising for his laptop, to just ensure that the tests pass on his machine, regardless of how they perform on ours. And then pray that this coincides with Travis as well otherwise we have to get creative to not get borken builds all the time.

Reply:
Ok let's make this really weird. On my mac both pass the test (I still have the rosenbrock fail, but rastigrin works fine), but on my linux machine I also have a fail for the static array and a pass for the `malloc`...